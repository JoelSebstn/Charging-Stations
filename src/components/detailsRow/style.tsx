import { StyleSheet } from "react-native";

export const Styles = StyleSheet.create({
    alignStart: { flex: 1, alignItems: 'flex-start' },
    textFont22: { fontSize: 22 },
    margin20: { marginBottom: 20 },
    flexRow: { flexDirection: 'row' },
});