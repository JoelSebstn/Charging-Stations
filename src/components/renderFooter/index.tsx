import { ActivityIndicator, Text, TouchableOpacity, View } from "react-native";
import { useNavigation } from "@react-navigation/native";


export function renderFooter(isLoading: any) {

    // Check If Loading
    if (isLoading) {
        return (
            <ActivityIndicator />
        )
    }
    else {
        return null;
    }

};