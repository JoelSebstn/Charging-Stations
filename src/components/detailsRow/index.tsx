import { View, Text } from "react-native";
import { Styles } from "./style";

export function DetailsRow({ rowName, data }: any) {
    return (
        <View style={[Styles.flexRow, Styles.margin20]}>
            <View style={Styles.alignStart}><Text style={Styles.textFont22}>{rowName}:</Text></View>
            <View style={Styles.alignStart}><Text style={Styles.textFont22}> {data}</Text></View>
        </View >)
}