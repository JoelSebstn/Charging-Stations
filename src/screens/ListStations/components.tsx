import { ActivityIndicator, Text, TouchableOpacity, View } from "react-native";
import { Styles } from "./style";
import { useNavigation } from "@react-navigation/native";

export function HeaderRow() {
    return (<><View style={Styles.flexRow}>
        <View style={Styles.alignCentre}><Text style={Styles.textFont22}>Name</Text></View>
        <View style={Styles.alignCentre}><Text style={Styles.textFont22}> Status</Text></View>
    </View>
        <View
            style={Styles.horizontalLine}
        /></>
    )
}
export function RenderItem(item: any) {
    const data = item["item"].data();
    const stationId = item["item"].data().id;
    const navigation: any = useNavigation();
    return (
        <TouchableOpacity onPress={() => navigation.navigate('DetailsSCene', { stationId: stationId })}>
            <View style={{ flexDirection: 'row', marginVertical: 12 }}>
                <View style={Styles.alignCentre}><Text style={Styles.textFont22}>{data.name}</Text></View>
                <View style={Styles.alignCentre}><Text style={Styles.textFont22}> {data.status}</Text></View>
            </View>
        </TouchableOpacity>
    )

}
export function renderFooter(isLoading: any) {
    try {
        // Check If Loading
        if (isLoading) {
            return (
                <ActivityIndicator />
            )
        }
        else {
            return null;
        }
    }
    catch (error) {
        console.log(error);
    }
};