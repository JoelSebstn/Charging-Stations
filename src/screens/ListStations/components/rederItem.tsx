import { useNavigation } from "@react-navigation/native";
import { TouchableOpacity, View, Text } from "react-native";
import { Styles } from "../style";

export function RenderItem(item: any) {
    const data = item["item"];
    const stationId = data.id;
    const navigation: any = useNavigation();
    return (
        <TouchableOpacity onPress={() => navigation.navigate('DetailsSCene', { stationId: stationId })}>
            <View style={Styles.rowMargin12}>
                <View style={Styles.alignCentre}><Text style={Styles.textFont22}>{data.name}</Text></View>
                <View style={Styles.alignCentre}><Text style={Styles.textFont22}> {data.status}</Text></View>
            </View>
        </TouchableOpacity>
    )

}