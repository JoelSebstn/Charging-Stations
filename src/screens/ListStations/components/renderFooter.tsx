import { ActivityIndicator, Text, TouchableOpacity, View } from "react-native";
import { Styles } from "../style";
import { useNavigation } from "@react-navigation/native";


export function renderFooter(isLoading: any) {
    try {
        // Check If Loading
        if (isLoading) {
            return (
                <ActivityIndicator />
            )
        }
        else {
            return null;
        }
    }
    catch (error) {
        console.log(error);
    }
};