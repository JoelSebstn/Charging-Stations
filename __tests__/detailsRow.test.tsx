import { render } from "@testing-library/react-native";
import { DetailsRow } from "../src/components/detailsRow";
import renderer from 'react-test-renderer';

describe("Test details ROw Componet", () => {
    it('Check contents', () => {
        const { getByText } = render(<DetailsRow rowName={'Status'} data={"Available"} />)
        const label = getByText("Available");
        expect(label).toBeTruthy();
    })
    it('renders the correct message', () => {
        const { queryByText } = render(<DetailsRow rowName={'Parking Access'} data={"Not Restricted"} />);
        expect(queryByText('Not Restricted')).not.toBeNull();
    });
    it("Matches Detail Snapshot", () => {
        const domTree = renderer.create(<DetailsRow rowName={'Status'} data={"Available"} />).toJSON();
        expect(domTree).toMatchSnapshot();
    });

})