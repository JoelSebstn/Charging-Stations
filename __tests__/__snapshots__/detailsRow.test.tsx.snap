// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test details ROw Componet Matches Detail Snapshot 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "marginBottom": 20,
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 22,
        }
      }
    >
      Status
      :
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 22,
        }
      }
    >
       
      Available
    </Text>
  </View>
</View>
`;
